java/test/jmri/jmrit/operations/rollingstock/engines/tools/ExportEnginesTest.java

```diff
@@ -1,33 +1,66 @@
 package jmri.jmrit.operations.rollingstock.engines.tools;
 
-import jmri.jmrit.operations.rollingstock.engines.tools.ExportEngines;
-import jmri.util.JUnitUtil;
+import java.awt.GraphicsEnvironment;
+import jmri.jmrit.operations.OperationsSwingTestCase;
+import jmri.util.JUnitOperationsUtil;
+import jmri.util.swing.JemmyUtil;
 import org.junit.After;
 import org.junit.Assert;
+import org.junit.Assume;
 import org.junit.Before;
 import org.junit.Test;
 
 /**
  *
  * @author Paul Bender Copyright (C) 2017	
  */
-public class ExportEnginesTest {
+public class ExportEnginesTest extends OperationsSwingTestCase {
 
     @Test
     public void testCTor() {
         ExportEngines t = new ExportEngines();
         Assert.assertNotNull("exists",t);
     }
+    
+    @Test
+    public void testCreateFile() {
+        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
+        
+        JUnitOperationsUtil.initOperationsData();
+        ExportEngines exportEngines = new ExportEngines();
+        Assert.assertNotNull("exists", exportEngines);
+
+        // should cause export complete dialog to appear
+        Thread export = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                exportEngines.writeOperationsEngineFile();
+            }
+        });
+        export.setName("Export Engines"); // NOI18N
+        export.start();
+
+        jmri.util.JUnitUtil.waitFor(() -> {
+            return export.getState().equals(Thread.State.WAITING);
+        }, "wait for prompt");
+
+        JemmyUtil.pressDialogButton(Bundle.getMessage("ExportComplete"), "OK");
+
+        java.io.File file = new java.io.File(ExportEngines.defaultOperationsFilename());
+        Assert.assertTrue("Confirm file creation", file.exists());
+    }
 
     // The minimal setup for log4J
+    @Override
     @Before
-    public void setUp() {
-        JUnitUtil.setUp();
+    public void setUp() throws Exception {
```

Comment:
It looks to me like OperationsSwingTestCase setUp and tearDown don't really need to throw Exception in their signature. It would be better to just remove that, then the subclasses wouldn't need to have a throws clause either.

Reply:
In the past the tests used to extend TestCase, which did throw exceptions.  Cleaned things up as requested.